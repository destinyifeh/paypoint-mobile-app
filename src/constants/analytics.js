export const ACCOUNT_OPENING_SAVE_CLICK = 'account_opening_save_click';
export const ACCOUNT_OPENING_SAVE_FAILURE = 'account_opening_save_failure';
export const ACCOUNT_OPENING_SAVE_SUCCESS = 'account_opening_save_success';
export const ACCOUNT_OPENING_SUBMIT_CLICK = 'account_opening_submit_click';
export const ACCOUNT_OPENING_SUBMIT_FAILURE = 'account_opening_submit_failure';
export const ACCOUNT_OPENING_SUBMIT_SUCCESS = 'account_opening_submit_success';

export const ACCOUNT_VALIDATION_CLICK = 'account_validation_click';
export const ACCOUNT_VALIDATION_FAILURE = 'account_validation_failure';
export const ACCOUNT_VALIDATION_SUCCESS = 'account_validation_success';

export const BILL_PAYMENT_INITIATE_CLICK = 'bill_payment_initiate_click';
export const BILL_PAYMENT_INITIATE_FAILURE = 'bill_payment_initiate_failure';
export const BILL_PAYMENT_INITIATE_SUCCESS = 'bill_payment_initiate_success';

export const BILL_PAYMENT_PROCEED_CLICK = 'bill_payment_proceed_click';
export const BILL_PAYMENT_PROCEED_FAILURE = 'bill_payment_proceed_failure';
export const BILL_PAYMENT_PROCEED_SUCCESS = 'bill_payment_proceed_success';

export const BROADCAST_MESSAGE_READ = 'broadcast_message_read';

export const BUY_AIRTIME_INITIATE_CLICK = 'buy_airtime_initiate_click';
export const BUY_AIRTIME_INITIATE_FAILURE = 'buy_airtime_initiate_failure';
export const BUY_AIRTIME_INITIATE_SUCCESS = 'buy_airtime_initiate_success';

export const BUY_AIRTIME_PROCEED_CLICK = 'buy_airtime_proceed_click';
export const BUY_AIRTIME_PROCEED_FAILURE = 'buy_airtime_proceed_failure';
export const BUY_AIRTIME_PROCEED_SUCCESS = 'buy_airtime_proceed_success';

export const CASH_IN_INITIATE_CLICK = 'cash_in_initiate_click';
export const CASH_IN_INITIATE_FAILURE = 'cash_in_initiate_failure';
export const CASH_IN_INITIATE_SUCCESS = 'cash_in_initiate_success';

export const CASH_IN_PROCEED_CLICK = 'cash_in_proceed_click';
export const CASH_IN_PROCEED_FAILURE = 'cash_in_proceed_failure';
export const CASH_IN_PROCEED_SUCCESS = 'cash_in_proceed_success';

export const DEVICE_AUTHENTICATION_CLICK = 'device_authentication_click';
export const DEVICE_AUTHENTICATION_FAILURE = 'device_authentication_failure';
export const DEVICE_AUTHENTICATION_SUCCESS = 'device_authentication_success';

export const FIRST_LOGIN_AFTER_APPROVAL_CLICK = 'first_login_after_approval';
export const FIRST_LOGIN_AFTER_APPROVAL_FAILURE = 'first_login_after_approval_failure';
export const FIRST_LOGIN_AFTER_APPROVAL_SUCCESS = 'first_login_after_approval_success';

export const FIRST_WALLET_FUNDING_AFTER_APPROVAL_CLICK = 'first_wallet_funding_after_approval_click';
export const FIRST_WALLET_FUNDING_AFTER_APPROVAL_FAILURE = 'first_wallet_funding_after_approval_failure';
export const FIRST_WALLET_FUNDING_AFTER_APPROVAL_SUCCESS = 'first_wallet_funding_after_approval_success';

export const IN_APP_FUNDING_INITIATE_CLICK = 'in_app_funding_initiate_click';
export const IN_APP_FUNDING_INITIATE_FAILURE = 'in_app_funding_initiate_failure';
export const IN_APP_FUNDING_INITIATE_SUCCESS = 'in_app_funding_initiate_success';
export const IN_APP_FUNDING_DROP_OFF = 'in_app_funding_drop_off';
export const IN_APP_FUNDING_PROCEED_CLICK = 'in_app_funding_proceed_click';
export const IN_APP_FUNDING_PROCEED_FAILURE = 'in_app_funding_proceed_failure';
export const IN_APP_FUNDING_PROCEED_SUCCESS = 'in_app_funding_proceed_success';

export const LOGIN_CLICK = 'login_click';
export const LOGIN_FAILURE = 'login_failure';
export const LOGIN_SUCCESS = 'login_success';

export const LOGIN_AFTER_SIGNUP_START = 'login_after_signup_start';
export const LOGIN_AFTER_SIGNUP_FAILURE = 'login_after_signup_failure';
export const LOGIN_AFTER_SIGNUP_SUCCESS = 'login_after_signup_success';

export const PURCHASE = 'purchase';

export const RATING_CLICK = 'rating_click';

export const SIGNUP_CLICK = 'signup_click';
export const SIGNUP_FAILURE = 'signup_failure';
export const SIGNUP_SUCCESS = 'signup_success';

export const TRANSACTION_INITIATE_CLICK = 'transaction_initiate_click';
export const TRANSACTION_INITIATE_FAILURE = 'transaction_initiate_failure';
export const TRANSACTION_INITIATE_SUCCESS = 'transaction_initiate_success';

export const TRANSACTION_PROCEED_CLICK = 'transaction_proceed_click';
export const TRANSACTION_PROCEED_FAILURE = 'transaction_proceed_failure';
export const TRANSACTION_PROCEED_SUCCESS = 'transaction_proceed_success';

export const TRANSFER_TO_ACCOUNT_INITIATE_CLICK = 'transaction_to_account_initiate_click';
export const TRANSFER_TO_ACCOUNT_INITIATE_FAILURE = 'transaction_to_account_initiate_failure';
export const TRANSFER_TO_ACCOUNT_INITIATE_SUCCESS = 'transaction_to_account_initiate_success';

export const TRANSFER_TO_ACCOUNT_PROCEED_CLICK = 'transaction_to_account_proceed_click';
export const TRANSFER_TO_ACCOUNT_PROCEED_FAILURE = 'transaction_to_account_proceed_failure';
export const TRANSFER_TO_ACCOUNT_PROCEED_SUCCESS = 'transaction_to_account_proceed_success';

export const TRANSFER_TO_AGENT_INITIATE_CLICK = 'transaction_to_agent_initiate_click';
export const TRANSFER_TO_AGENT_INITIATE_FAILURE = 'transaction_to_agent_initiate_failure';
export const TRANSFER_TO_AGENT_INITIATE_SUCCESS = 'transaction_to_agent_initiate_success';

export const TRANSFER_TO_AGENT_PROCEED_CLICK = 'transaction_to_agent_proceed_click';
export const TRANSFER_TO_AGENT_PROCEED_FAILURE = 'transaction_to_agent_proceed_failure';
export const TRANSFER_TO_AGENT_PROCEED_SUCCESS = 'transaction_to_agent_proceed_success';

export const WITHDRAW_INITIATE_CLICK = 'withdraw_initiate_click';
export const WITHDRAW_INITIATE_FAILURE = 'withdraw_initiate_failure';
export const WITHDRAW_INITIATE_SUCCESS = 'withdraw_initiate_success';

export const WITHDRAW_PROCEED_CLICK = 'withdraw_proceed_click';
export const WITHDRAW_PROCEED_FAILURE = 'withdraw_proceed_failure';
export const WITHDRAW_PROCEED_SUCCESS = 'withdraw_proceed_success';

export const SCREEN_VIEW = 'screen_view';

export const POS_BILL_PAYMENT_INITIATE_CLICK = 'POS_BILL_PAYMENT_INITIATE_CLICK'
export const POS_BILL_PAYMENT_INITIATE_FAILURE = 'POS_BILL_PAYMENT_INITIATE_FAILURE'
export const POS_BILL_PAYMENT_INITIATE_SUCCESS = 'POS_BILL_PAYMENT_INITIATE_SUCCESS'
export const POS_BILL_PAYMENT_PROCEED_CLICK = 'POS_BILL_PAYMENT_PROCEED_CLICK'
export const POS_BILL_PAYMENT_PROCEED_FAILURE = 'POS_BILL_PAYMENT_PROCEED_FAILURE'
export const POS_BILL_PAYMENT_PROCEED_SUCCESS = 'POS_BILL_PAYMENT_PROCEED_SUCCESS'

export const POS_BUY_AIRTIME_INITIATE_CLICK = 'POS_BUY_AIRTIME_INITIATE_CLICK'
export const POS_BUY_AIRTIME_INITIATE_FAILURE = 'POS_BUY_AIRTIME_INITIATE_FAILURE'
export const POS_BUY_AIRTIME_INITIATE_SUCCESS = 'POS_BUY_AIRTIME_INITIATE_SUCCESS'
export const POS_BUY_AIRTIME_PROCEED_CLICK = 'POS_BUY_AIRTIME_PROCEED_CLICK'
export const POS_BUY_AIRTIME_PROCEED_FAILURE = 'POS_BUY_AIRTIME_PROCEED_FAILURE'
export const POS_BUY_AIRTIME_PROCEED_SUCCESS = 'POS_BUY_AIRTIME_PROCEED_SUCCESS'

export const POS_CASH_IN_INITIATE_CLICK = 'POS_CASH_IN_INITIATE_CLICK'
export const POS_CASH_IN_INITIATE_FAILURE = 'POS_CASH_IN_INITIATE_FAILURE'
export const POS_CASH_IN_INITIATE_SUCCESS = 'POS_CASH_IN_INITIATE_SUCCESS'
export const POS_CASH_IN_PROCEED_CLICK = 'POS_CASH_IN_PROCEED_CLICK'
export const POS_CASH_IN_PROCEED_FAILURE = 'POS_CASH_IN_PROCEED_FAILURE'
export const POS_CASH_IN_PROCEED_SUCCESS = 'POS_CASH_IN_PROCEED_SUCCESS'

export const POS_TRANSACTION_INITIATE_CLICK = 'POS_TRANSACTION_INITIATE_CLICK'
export const POS_TRANSACTION_INITIATE_FAILURE = 'POS_TRANSACTION_INITIATE_FAILURE'
export const POS_TRANSACTION_INITIATE_SUCCESS = 'POS_TRANSACTION_INITIATE_SUCCESS'

export const POS_TRANSACTION_PROCEED_CLICK = 'POS_TRANSACTION_PROCEED_CLICK'
export const POS_TRANSACTION_PROCEED_FAILURE = 'POS_TRANSACTION_PROCEED_FAILURE'
export const POS_TRANSACTION_PROCEED_SUCCESS = 'POS_TRANSACTION_PROCEED_SUCCESS'

export const POS_TRANSFER_TO_ACCOUNT_INITIATE_CLICK = 'POS_TRANSFER_TO_ACCOUNT_INITIATE_CLICK'
export const POS_TRANSFER_TO_ACCOUNT_INITIATE_FAILURE = 'POS_TRANSFER_TO_ACCOUNT_INITIATE_FAILURE'
export const POS_TRANSFER_TO_ACCOUNT_INITIATE_SUCCESS = 'POS_TRANSFER_TO_ACCOUNT_INITIATE_SUCCESS'
export const POS_TRANSFER_TO_ACCOUNT_PROCEED_CLICK = 'POS_TRANSFER_TO_ACCOUNT_PROCEED_CLICK'
export const POS_TRANSFER_TO_ACCOUNT_PROCEED_FAILURE = 'POS_TRANSFER_TO_ACCOUNT_PROCEED_FAILURE'
export const POS_TRANSFER_TO_ACCOUNT_PROCEED_SUCCESS = 'POS_TRANSFER_TO_ACCOUNT_PROCEED_SUCCESS'

export const POS_TRANSFER_TO_AGENT_INITIATE_CLICK = 'POS_TRANSFER_TO_AGENT_INITIATE_CLICK'
export const POS_TRANSFER_TO_AGENT_INITIATE_FAILURE = 'POS_TRANSFER_TO_AGENT_INITIATE_FAILURE'
export const POS_TRANSFER_TO_AGENT_INITIATE_SUCCESS = 'POS_TRANSFER_TO_AGENT_INITIATE_SUCCESS'
export const POS_TRANSFER_TO_AGENT_PROCEED_CLICK = 'POS_TRANSFER_TO_AGENT_PROCEED_CLICK'
export const POS_TRANSFER_TO_AGENT_PROCEED_FAILURE = 'POS_TRANSFER_TO_AGENT_PROCEED_FAILURE'
export const POS_TRANSFER_TO_AGENT_PROCEED_SUCCESS = 'POS_TRANSFER_TO_AGENT_PROCEED_SUCCESS'

export const POS_WITHDRAW_INITIATE_CLICK = 'POS_WITHDRAW_INITIATE_CLICK'
export const POS_WITHDRAW_INITIATE_FAILURE = 'POS_WITHDRAW_INITIATE_FAILURE'
export const POS_WITHDRAW_INITIATE_SUCCESS = 'POS_WITHDRAW_INITIATE_SUCCESS'
export const POS_WITHDRAW_PROCEED_CLICK = 'POS_WITHDRAW_PROCEED_CLICK'
export const POS_WITHDRAW_PROCEED_FAILURE = 'POS_WITHDRAW_PROCEED_FAILURE'
export const POS_WITHDRAW_PROCEED_SUCCESS = 'POS_WITHDRAW_PROCEED_SUCCESS'
