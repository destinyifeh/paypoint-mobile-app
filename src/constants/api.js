export const DELETE_METHOD = "DELETE";
export const GET_METHOD = "GET";
export const POST_METHOD = "POST";
export const PUT_METHOD = "PUT";

export const COMPLETED_STATUS = "COMPLETED";
export const ERROR_STATUS = "ERROR";
export const FAILED_STATUS = "Failed";
export const PENDING_STATUS = "Pending";
export const SUCCESS_STATUS = "SUCCESS";
export const SUCCESSFUL_STATUS = "Successful";

export const HTTP_ACCEPTED = 202;
export const HTTP_CONFLICT = 409;
export const HTTP_CREATED = 201;
export const HTTP_FORBIDDEN = 403;
export const INSUFFICIENT_FUND = "40006";
export const HTTP_INTERNAL_SERVER_ERROR = 500;
export const HTTP_METHOD_NOT_ALLOWED = 405;
export const HTTP_NOT_FOUND = 404;
export const HTTP_OK = 200;

export const DEFAULT_API_ERROR_MESSAGE =
  "Something went wrong! Please, repeat that last step again.";
