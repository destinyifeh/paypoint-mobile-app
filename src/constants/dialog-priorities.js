export const BLOCKER = 'BLOCKER';
export const CASUAL = 'CASUAL';
export const UNIMPORTANT = 'UNIMPORTANT';
