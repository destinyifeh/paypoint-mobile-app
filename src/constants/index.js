import { Dimensions } from "react-native";
import { COLOUR_GREEN, COLOUR_ORANGE, COLOUR_RED } from "./styles";

export const ACCESS_TOKEN = "access_token";
export const AGENT = "agent";
export const AGENT_TYPE_ID = 4;
export const APP_NAME = 'Quickteller Paypoint';
export const APPLICANT = 'applicant';
export const APPLICATION = 'application';
export const APPLICATION_ASSISTED = 'application';
export const APPROVED = 'APPROVED';
export const AUTO_LOGIN_FAILED_MESSAGE = 'You already have an Interswitch account. Login with existing credentials!';
export const DEFAULT_PAGE_SIZE = 10;
export const DEFAULT_DOMAIN_CODE = "APP";
export const HAS_USER_RATED_APP_ON_THE_STORE =
  "HAS_USER_RATED_APP_ON_THE_STORE";
export const LOGIN_CREDENTIALS = "LOGIN_CREDENTIALS";
export const USER_PROFILES = "USER_PROFILES";
export const SUPER_AGENT = "super agent";
export const USER = "user";
export const BVN_STATUS = "bvn_status";

export const COPIED_TO_CLIPBOARD = "Copied to clipboard!";

export const EMPTY_AMOUNT = "0.00";

export const FILE_UPLOAD_LIMIT = 1000000;
export const FILE_UPLOAD_LIMIT_MESSAGE =
  "The file you uploaded exceeds the file limit of 4MB.";

export const AWAITING_VALIDATION_APPLICATION_STATUS = "0";
export const AWAITING_APPROVAL_APPLICATION_STATUS = "1";
export const APPROVED_APPLICATION_STATUS = "2";
export const DECLINED_APPLICATION_STATUS = "3";

export const SHOW_STATUS_BAR = "SHOW_STATUS_BAR";
export const CANNOT_ACCESS_DASHBOARD =
  "You are not authorized to access this dashboard.";
export const IFIS_K3 = "ifis-k3";

export const EXPIRED_SESSION_MESSAGE =
  "Your session has expired. Please, re-login.";
export const LOGIN_DETAILS = "login_details";
export const INVALID_FORM_MESSAGE =
  "There are some errors in the form. Please, scroll through the form, resolve all invalid fields and re-submit.";

export const TRANSACTION_CURRENCY = "NGN";
export const TRANSACTION_FILTERS = "Transaction Filters";
export const TRANSFER_PAYMENT_ITEM_CODE = "10428";

export const DRAFT = "draft";

export const MOBILE = "mobile";
export const MONTHS = [
  "Jan",
  "Feb",
  "Mar",
  "Apr",
  "May",
  "Jun",
  "Jul",
  "Aug",
  "Sep",
  "Oct",
  "Nov",
  "Dec",
];
export const MONTHS_NUMBERS = [
  "01",
  "02",
  "03",
  "04",
  "05",
  "06",
  "07",
  "08",
  "09",
  "10",
  "11",
  "12",
];

export const NIGERIA = "Nigeria";
export const NIGERIA_SHORT_CODE = "NG";
export const MAX_SELECT_BOX_RECORDS = 300;
export const OTP_SUCCESSFULLY_SENT = "OTP has been successfully sent";
export const SENDING = "Sending";
export const SUCCESS = "Success";

export const AIRTIME_AND_DATA = "airtime-and-data";
export const BILLS = "bills";
export const CASH_IN = "cash-in";
export const CASH_OUT = "cash_out";
export const MMO = "mmo";
export const PAY_A_BILL = "pay-a-bill";
export const RECHARGE = "recharge";
export const SEND_MONEY = "send-money";
export const TRANSFER = "transfer";
export const FUND_TRANSFER_UNIFIED = "FUND_TRANSFER_UNIFIED";
export const CASHIN = "TRANSFER_CASH_IN";
export const TRANSFER_TO_ACCOUNT = "transfer-to-account";
export const TRANSFER_TO_AGENT = "transfer-to-agent";
export const WITHDRAW = "withdraw";
export const PAYCODE = "paycode";
export const USSD = "ussd";
export const WITHDRAW_VIA_PAYCODE = "withdraw_via_paycode";
export const WITHDRAW_VIA_USSD = "withdraw_via_ussd";

export const PENDING_SCENE_AFTER_APPLICATION_SUBMISSION =
  "PENDING_SCENE_AFTER_APPLICATION_SUBMISSION";
export const PENDING_SIGNUP = "pending-signup";

export const SHARE_RECEIPT_MESSAGE = "Quickteller Paypoint Transaction Receipt";
export const SHARE_RECEIPT_TITLE = "Quickteller Paypoint";

export const TRANSACTION_NARRATION_MAX_LENGTH = 20;

export const WALLET = "wallet";

export const CONTACT_US_EMAIL = "ifiscustomercare@interswitchgroup.com";
export const CONTACT_US_WEBPAGE = "https://help.interswitchgroup.com/";
export const CONTACT_US_PHONE = "07009065000";

export const PENDING_FUND_WALLET_MESSAGE =
  "Your transaction is currently being processed. Check your transaction history report for the latest status of the transaction.";
export const SUCCESSFUL_FUND_WALLET_MESSAGE =
  "Your wallet has been funded successfully!";
export const SUCCESSFUL_CARD_LINKING_MESSAGE = "Card was linked successfully!";

export const TRANSACTION_ERROR = "Transaction Error";

export const TRANSACTION_STATUS_CODE_COLORS = {
  Successful: COLOUR_GREEN,
  Failed: COLOUR_RED,
  Pending: COLOUR_ORANGE,
};

export const PENDING_TRANSACTION_STATUS_CODE = "02";

export const TRANSACTION_STATUS_COLORS = {
  "00": COLOUR_GREEN,
  "01": COLOUR_RED,
  "02": COLOUR_ORANGE,
  "03": COLOUR_RED,
  "04": COLOUR_RED,
};

export const TRANSACTION_STATUS_LABELS = {
  "00": "Successful",
  "01": "Failed",
  "02": "Pending",
  "03": "Duplicate",
  "04": "Not Found",
};

export const BUY_AIRTIME = "Buy Airtime";
export const PAY_A_BILL_NORMALIZED = "Pay a Bill";

export const NO_AUTH_SCREENS = [
  "GuestLogin",
  "Login",
  "Logout",
  "ResetPassword",
  "Signup",
  "Welcome",
  "Landing",
  "VerifyPhone",
];

export const MIME_TYPE_CSV = "text/csv";

export const SEEN_FEATURES = "SEEN_FEATURES";

export const WINDOW_WIDTH = Dimensions.get("window").width;
