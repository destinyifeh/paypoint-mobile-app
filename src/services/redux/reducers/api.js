const initialState = {
  data: {},
  isFetching: false,
  isError: false
}

export default function asyncReducer(state=initialState, action) {
  return;
}
