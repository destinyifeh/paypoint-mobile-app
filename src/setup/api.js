import AccountOpening from "../services/api/resources/account-opening";
import ApiGateway from "../services/api/resources/api-gateway";
import Audit from "../services/api/resources/audit";
import BankMonitoring from "../services/api/resources/bank-monitoring";
import Catalog from "../services/api/resources/catalog";
import CRMService from "../services/api/resources/crm";
import Liquidity from "../services/api/resources/liquidity";
import Messaging from "../services/api/resources/messaging";
import Nip from "../services/api/resources/nip";
import Onboarding from "../services/api/resources/onboarding";
import Passport from "../services/api/resources/passport";
import Platform from "../services/api/resources/platform";
import Quickteller from "../services/api/resources/quickteller";
import Settlement from "../services/api/resources/settlement";
import Transaction from "../services/api/resources/transaction";
import TransactionHistory from "../services/api/resources/transaction-history";
import TransactionV1 from "../services/api/resources/transaction-v1";
import TransactionV3 from "../services/api/resources/transaction-v3";
import UserManagement from "../services/api/resources/user-management";
export const accountOpeningService = new AccountOpening();
export const apiGatewayService = new ApiGateway();
export const auditService = new Audit();
export const liquidityService = new Liquidity();
export const messagingService = new Messaging();
export const nipService = new Nip();
export const onboardingService = new Onboarding();
export const passportService = new Passport();
export const platformService = new Platform();
export const quicktellerService = new Quickteller();
export const catalogService = new Catalog();
export const settlementService = new Settlement();
export const transactionHistoryService = new TransactionHistory();
export const transactionService = new Transaction();
export const transactionServiceV3 = new TransactionV3();
export const userManagementService = new UserManagement();
export const transactionServiceV1 = new TransactionV1();
export const bankMonitoringService = new BankMonitoring();
export const crmService = new CRMService();
