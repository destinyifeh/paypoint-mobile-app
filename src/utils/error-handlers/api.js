import NetInfo from '@react-native-community/netinfo';
import {
  DEFAULT_API_ERROR_MESSAGE,
  HTTP_INTERNAL_SERVER_ERROR,
} from "../../constants/api";
import ErrorTemplates from "../../fixtures/error_templates";
import { handleConnectionStatusWrapper } from "../../setup/background-tasks";

export class ApiErrorHandler {

  _findTemplateForErrorResponse(parsedErrorResponse) {
    // TODO refactor this to cater for `mode` and other parameters
    return ErrorTemplates.find(
      (value) => value.message_from_api === parsedErrorResponse.message
    );
  }

  _parseApiErrorResponse(rawApiResponse) {
    const errorObject = {
      code: null,
      message: null,
      message_to_user: null,
    };
    if (!rawApiResponse) {
      return errorObject;
    }

    errorObject.code = rawApiResponse.code;

    // pick out the message in the response object.
    errorObject.message =
      rawApiResponse.description ||
      rawApiResponse.errorMessage ||
      rawApiResponse.message;
    if (rawApiResponse.responseData && rawApiResponse.responseData.length > 0) {
      let messageList = rawApiResponse.description.toUpperCase() + "\n\n";
      rawApiResponse.responseData.forEach((error) => {
        const fieldName = error.fieldName.split(".");
        let userFriendlyField = fieldName[fieldName.length - 1];
        userFriendlyField =
          userFriendlyField[0].toUpperCase() + userFriendlyField.slice(1);
        messageList += `${userFriendlyField} ${error.message}\n`;
      });
      errorObject.message = messageList;
      errorObject.message_to_user = messageList;
    } else if (
      rawApiResponse.responseData &&
      rawApiResponse.responseData.description
    ) {
      errorObject.message = rawApiResponse.responseData.description;
    } else if (rawApiResponse.errors && rawApiResponse.errors.length > 0) {
      errorObject.message =
        rawApiResponse.errors[0].errorMessage ||
        rawApiResponse.errors[0].message;
    } else if (rawApiResponse.error && rawApiResponse.error.length > 0) {
      errorObject.message = DEFAULT_API_ERROR_MESSAGE;
    }
    return errorObject;
  }

  handleApiErrorResponse(
    rawApiResponse,
    errorMessageKeywords,
    forceShowUnknownErrors
  ) {
    const parsedErrorResponse = this._parseApiErrorResponse(rawApiResponse);

    const defaultErrorMessage = forceShowUnknownErrors
      ? parsedErrorResponse.message
      : DEFAULT_API_ERROR_MESSAGE;
    let errorMessage = rawApiResponse.code?.startsWith(
      JSON.stringify(HTTP_INTERNAL_SERVER_ERROR)
    )
      ? defaultErrorMessage
      : this._findTemplateForErrorResponse(parsedErrorResponse)
        ?.message_to_user;

    if (!errorMessage) {
      errorMessage = parsedErrorResponse.message;
    }

    if (errorMessageKeywords) {
      let refinedErrorMessage = errorMessage;

      Object.keys(errorMessageKeywords).map((keyword) => {
        refinedErrorMessage = refinedErrorMessage
          ? refinedErrorMessage.replace(
            `{${keyword}}`,
            errorMessageKeywords[keyword]
          )
          : null;
      });

      return refinedErrorMessage;
    }

    return errorMessage || DEFAULT_API_ERROR_MESSAGE;
  }
}

const apiErrorHandler = new ApiErrorHandler();

async function checkNetwork (message) {
  let val = message;
  try {
    const connectionState = await NetInfo.fetch();
    val = handleConnectionStatusWrapper(connectionState);
    return val === ""? message : val;
  } catch (e) {}
  return val;
}

export default async function handleErrorResponse(
  responseObj,
  errorMessageKeywords,
  forceShowUnknownErrors
) {


  let message = apiErrorHandler.handleApiErrorResponse(
    responseObj,
    errorMessageKeywords,
    forceShowUnknownErrors
  );

  if (message === DEFAULT_API_ERROR_MESSAGE || message === "An error occured. Please, try again later!") {
    return await checkNetwork(message)
  }

  return message.length > 80 ? DEFAULT_API_ERROR_MESSAGE : message;

  // check network here if it is a default error message


}
