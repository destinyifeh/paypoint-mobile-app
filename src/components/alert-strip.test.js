'use strict';

import React from 'react';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import AlertStrip from './alert-strip';
import { ERROR_STATUS, SUCCESS_STATUS } from '../constants/api';

configure({
  adapter: new Adapter()
});

it('Error AlertStrip renders correctly', () => {
  const tree = renderer
    .create(<AlertStrip 
      variant={ERROR_STATUS} 
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Success AlertStrip renders correctly', () => {
  const tree = renderer
    .create(<AlertStrip 
      variant={SUCCESS_STATUS} 
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
