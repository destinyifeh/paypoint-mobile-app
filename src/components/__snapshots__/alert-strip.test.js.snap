// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error AlertStrip renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 4,
        "borderWidth": 2,
        "color": "#3A3B3B",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
      },
      Object {
        "backgroundColor": "#EE312A90",
        "borderColor": "#EE312A",
      },
    ]
  }
>
  <View>
    <Text
      black={true}
      bold={true}
      style={
        Object {
          "color": "#3A3B3B",
          "fontWeight": "bold",
        }
      }
    >
      ERROR
    </Text>
    <Text
      black={true}
      style={
        Object {
          "color": "#3A3B3B",
        }
      }
    />
  </View>
</View>
`;

exports[`Success AlertStrip renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 4,
        "borderWidth": 2,
        "color": "#3A3B3B",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
      },
      undefined,
    ]
  }
>
  <View>
    <Text
      black={true}
      bold={true}
      style={
        Object {
          "color": "#3A3B3B",
          "fontWeight": "bold",
        }
      }
    >
      SUCCESS
    </Text>
    <Text
      black={true}
      style={
        Object {
          "color": "#3A3B3B",
        }
      }
    />
  </View>
</View>
`;
